File: frontend/types/nutrition.ts
export type MealType = 'breakfast' | 'lunch' | 'dinner' | 'snack'

export interface NutritionEntry {
  id: string
  user_id: string
  meal_type: MealType
  food_name: string
  quantity: number
  unit: string
  calories: number
  protein_g: number | null
  carbs_g: number | null
  fat_g: number | null
  fiber_g: number | null
  notes: string | null
  logged_at: string
  is_shared: boolean
  created_at: string
}

export interface NutritionDailySummary {
  date: string
  total_calories: number
  total_protein_g: number
  total_carbs_g: number
  total_fat_g: number
  entry_count: number
  entries: NutritionEntry[]
}

export interface CreateNutritionRequest {
  meal_type: MealType
  food_name: string
  quantity: number
  unit: string
  calories: number
  protein_g?: number
  carbs_g?: number
  fat_g?: number
  fiber_g?: number
  notes?: string
  logged_at?: string
  is_shared?: boolean
}

export interface NutritionListParams {
  skip?: number
  limit?: number
  start_date?: string
  end_date?: string
  meal_type?: MealType
}

File: backend/app/schemas/nutrition.py
from __future__ import annotations

import uuid
from datetime import date, datetime

from pydantic import BaseModel, Field

from app.core.constants import MealType


class NutritionCreateRequest(BaseModel):
    entry_date: date
    meal_type: MealType
    food_name: str = Field(min_length=1, max_length=200)
    calories: float = Field(ge=0, le=10000)
    protein_g: float | None = Field(default=None, ge=0, le=1000)
    carbs_g: float | None = Field(default=None, ge=0, le=1000)
    fat_g: float | None = Field(default=None, ge=0, le=1000)
    is_shared: bool = False


class NutritionResponse(BaseModel):
    id: uuid.UUID
    entry_date: date
    meal_type: str
    food_name: str
    calories: float
    protein_g: float | None
    carbs_g: float | None
    fat_g: float | None
    is_shared: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class NutritionListResponse(BaseModel):
    items: list[NutritionResponse]
    total: int
    skip: int
    limit: int


class DailySummaryResponse(BaseModel):
    date: date
    total_calories: float
    total_protein_g: float
    total_carbs_g: float
    total_fat_g: float
    entry_count: int
